src/app/space/app-launcher/services/app-launcher-gitprovider.service.ts

```diff
@@ -142,53 +135,33 @@ export class AppLauncherGitproviderService implements GitProviderService {
    * @returns {Observable<any>} list of existing GitHub repos
    */
   getGitHubRepoList(org: string): Observable<any> {
-    let url = this.END_POINT + this.API_BASE + 'repositories';
-    let location = org + '/';
-    if (this.gitHubUserLogin !== org) {
-      url += '?organization=' + org;
+    return this.getRepositories(org).pipe(map(AppLauncherGitproviderService.removeOrganizationPrefix));
+  }
+
+  private getRepositories(org: string = ''): Observable<string[]> {
+    if (this.repositories[org]) {
+      return of(this.repositories[org]);
     }
-    return this.http
-      .get(url, { headers: this.headers }).pipe(
-      map((resp: HttpResponse<any>) => {
-          let repoList = [];
-          if (resp) {
-            let responseList: string[] = resp as any;
-            responseList.forEach(function(ele) {
-              repoList.push(ele.replace(location, ''));
-            });
-          }
-          return repoList;
-        }),
+    return this.http.get(this.createUrl(org), { headers: this.headers }).pipe(
+      map((json) => json ? json as string[] : []),
+      map((repositories) => this.repositories[org] = repositories),
       catchError((error: HttpErrorResponse) => {
         return observableThrowError(error);
       }));
   }
 
   // Private
-
-  private getRequestParam(name: string): string {
-    let param = (new RegExp('[?&]' + encodeURIComponent(name) + '=([^&]*)')).exec(window.location.search);
-    if (param !== null) {
-      return decodeURIComponent(param[1]);
-    }
-    return null;
+  private static removeOrganizationPrefix(repositories: string[]): string[] {
+    return repositories.map((ele) => ele.replace(new RegExp('^[^/]+/'), ''));
   }
 
-  private redirectToAuth(url: string) {
-    window.location.href = url;
-  }
-
-  private handleError(error: HttpErrorResponse | any) {
-    // In a real world app, we might use a remote logging infrastructure
-    let errMsg: string;
-    if (error instanceof HttpResponse) {
-      const body = error.body || '';
-      const err = body.error || JSON.stringify(body);
-      errMsg = `${error.status} - ${error.statusText || ''} ${err}`;
-    } else {
-      errMsg = error.message ? error.message : error.toString();
+  private createUrl(org: string) {
+    // const url = this.END_POINT + this.API_BASE + 'repositories';
```

Comment:
could you remove these comments

Reply:
Removed comment here https://github.com/fabric8-ui/fabric8-ui/pull/3393/commits/1106b0cec2c03bcaf9c26c89e7080f4b26cfdc4b